src/include/cephfs/libcephfs.h

```diff
@@ -604,6 +604,23 @@ int ceph_readdir_r(struct ceph_mount_info *cmount, struct ceph_dir_result *dirp,
 int ceph_readdirplus_r(struct ceph_mount_info *cmount, struct ceph_dir_result *dirp, struct dirent *de,
 		       struct ceph_statx *stx, unsigned want, unsigned flags, struct Inode **out);
 
+/** FIXME: update the comments!!!!
+ * Get the next entry in an open snapshot delta's directory.
+ *
+ * @param cmount the ceph mount handle to use for performing the readdir.
+ * @param dirp the directory stream pointer from an opendir holding the state of the
+ *        next entry to return.
+ * @param snap_other - snapshot id to build delta
+ * @returns the next directory entry or NULL if at the end of the directory (or the directory
+ *          is empty.  This pointer should not be freed by the caller, and is only safe to
+ *          access between return and the next call to ceph_readdir or ceph_closedir.
+ */
+int ceph_readdir_snapdiff(struct ceph_mount_info* cmount,
```

Comment:
It's looks a bit unusual to pass in the snap-id of the _other_ snapshot against which the delta has to be built. Could this be the snapshot name?

FWIW, this interface can accept two directory stream pointers (one for each snapshot) and a path for which the entries are to be built.

Reply:
@ifed01 @batrick I'm open to suggestions on how the API would look like? Basically, I would like to avoid passing in snap-ids if possible.

Reply:
@vshankar - I've reworked the API to accept snapshot root(!) dir stream pointer instead of snapid.
There is still a snapid returned by the function to distinguish removed and existing entries. Probably we can get rid off it as well and replace with an explicit bool flag... Do you think we need should do that?

Reply:
@batrick - I refactored this part a bit, please take a look. Is that what you suggested?